source/config.js

```diff
@@ -235,6 +239,25 @@ if (typeof currentRootPath !== 'string') {
 }
 module.exports.rootPath = currentRootPath;
 
+let currentGitCommand = module.exports.gitCommand;
+if (typeof currentGitCommand !== 'string') {
+  currentGitCommand = '';
+} else if (currentGitCommand !== '') {
+  // must start with a slash
+  if (currentGitCommand.charAt(0) !== '/') {
+    currentGitCommand = '/' + currentGitCommand;
+  }
+  // can not end with a trailing slash - it should be a command
+  if (currentGitCommand.charAt(currentGitCommand.length - 1) === '/') {
```

Comment:
> Furthermore, considering this is a crux of all operations, I would like to reject on invalid inputs rather than adjust on invalid input as well to make the intent explicit.

In current patch set I left this part as is - how the setting should be checked? I see it like:
* check if a file exists
* check if it is a file, not a directory
* throw exception otherwise

And surely get rid of those adjustments.